### PR TITLE
Main

### PR DESCRIPTION
**Pull request by stivka**

### CODE REVIEW

File: probablility.py [hunk 1]
```diff
@@ -80,4 +80,4 @@ def printInGrid(numbers):
             print(numbers[i], end="  ")
     print('\n\n')
 
-playNumberOfTimes(100)
+playNumberOfTimes(100000)
```
Comment (MarkusTarn) [inline probablility.py]: It looks weird to hardcode this magic number.
Wouldn’t it make more sense to read it from command line argument?

`playNumberOfTimes(sys.argv[1])`

https://www.geeksforgeeks.org/command-line-arguments-in-python/amp/

File: probablility.py [hunk 2]
```diff
@@ -42,12 +42,12 @@ def play(playersQueue, numbersInsideBoxes):
                 # print('Player ' + str(playerNumber) + ' found their number after opening ' + str(numberOfBoxesOpened) + ' boxes!')
                 break
             elif numberOfBoxesOpened == 50:
-                print('Player ' + str(playerNumber) + ' did not find their number while opening ' + str(numberOfBoxesOpened) + ' boxes. All players shall die!')
+                #print('Player ' + str(playerNumber) + ' did not find their number while opening ' + str(numberOfBoxesOpened) + ' boxes. All players shall die!')
```
Comment (MarkusTarn) [inline probablility.py]: You know what would be megacool 😎
Check if VERBOSE env var is enabled and only then print verbose logs.
This way you don’t have to commit commented out code to git 😌

```
if (os.environ['VERBOSE']):
    print(…
```

then you can just run the script with verbose mode to see the verbose logs
`VERBOSE=true python3 probability.py 10000`

https://able.bio/rhett/how-to-set-and-get-environment-variables-in-python--274rgt5

**Review by MarkusTarn — CHANGES_REQUESTED** [review]

**Comment by MarkusTarn** [comment]
👌 Looks good to me, Approved!


On a personal taste of preference, I would not have added logging level as a second argument.
The reason being, that argument should be something that will have direct impact on the outcome result, where as logging level is more like a configuration parameter.

There are two ways that this is usually handled:
1) You read configuration parameters from environmental variables.
In this case I would change the line 5 - 6 with this one: `verboseLogging = os.environ['LOG_LEVEL'] =='verbose'`
And then later in the code you just refer it to: `if (verboseLogging): ...`
NB! Declaring it separately serves only the clean code purpose, nothing else.

Therefore, if you wish to run the script with verbose logging, you can do it by either one of the following solutions:
* Adding variable when running in command line `LOG_LEVEL=verbose py probability.py 1000`
* Add `LOG_LEVEL=verbose` to .env file
* Add the env var to VS code python runtime configuration so it's added when you press run button

2) Add it as a configuration flag to cli command instead of argument.
The difference between this one and the existing one would be, that it would be run by sth like 
`py probability.py 1000 --verbose`.
In this case you would replace lines 5-6 with this single line:
`verboseLogging = '--verbose' in sys.argv`

I would also choose either one, but probably not them both. Either way this solution works fine and changing it would probably be considered over engineering.

**Review by MarkusTarn — APPROVED** [review]

**Comment by stivka** [comment]
> 

Thanks for the thorough review, I really enjoyed it. Will implement option 1 for logging. Now that you said it, I agree that arguments should be used for function rather than utility. Altho option 2 is what I intended towards, to the effect that anyone that clones the project could immediately raise the log levels without really digging into the code. But option 1 is perhaps the more classic option and better to have space for other arguments ;)

**Pull request merged**